packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-header/pipeline-ai.spec.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import type { ComponentProps } from 'react';
+import { cleanup, fireEvent, render, screen } from '@testing-library/react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import type { SinonSpy } from 'sinon';
+import { Provider } from 'react-redux';
+
+import { PipelineAI } from './pipeline-ai';
+import configureStore from '../../../../test/configure-store';
+import {
+  AIPipelineActionTypes,
+  changeAIPromptText,
+} from '../../../modules/pipeline-builder/pipeline-ai';
+
+const noop = () => {
+  /* no op */
+};
+
+const renderPipelineAI = ({
+  ...props
+}: Partial<ComponentProps<typeof PipelineAI>> = {}) => {
+  const store = configureStore();
+
+  render(
+    <Provider store={store}>
+      <PipelineAI onClose={noop} show {...props} />
+    </Provider>
+  );
+  return store;
+};
+
+const feedbackPopoverTextAreaId = 'feedback-popover-textarea';
+
+describe('PipelineAI Component', function () {
+  let store: ReturnType<typeof configureStore>;
+  afterEach(cleanup);
+
+  describe('when rendered', function () {
+    let onCloseSpy: SinonSpy;
+    beforeEach(function () {
+      onCloseSpy = sinon.spy();
+      store = renderPipelineAI({
+        onClose: onCloseSpy,
+      });
+    });
+
+    it('calls to close robot button is clicked', function () {
+      expect(onCloseSpy.called).to.be.false;
+      const closeButton = screen.getByTestId('close-ai-button');
+      expect(closeButton).to.be.visible;
+      closeButton.click();
+      expect(onCloseSpy.calledOnce).to.be.true;
+    });
+  });
+
+  describe('when rendered with text', function () {
+    beforeEach(function () {
+      store = renderPipelineAI();
+      store.dispatch(changeAIPromptText('test'));
+    });
+
+    it('calls to clear the text when the X is clicked', function () {
+      expect(store.getState().pipelineBuilder.aiPipeline.aiPromptText).to.equal(
+        'test'
+      );
+
+      const clearTextButton = screen.getByTestId('ai-text-clear-prompt');
+      expect(clearTextButton).to.be.visible;
+      clearTextButton.click();
+
+      expect(store.getState().pipelineBuilder.aiPipeline.aiPromptText).to.equal(
+        ''
+      );
+    });
+  });
+
+  describe('Query AI Feedback', function () {
```

Comment:
Will update this to bring in the changes we did to the ai query feedback tests in https://github.com/mongodb-js/compass/pull/4735/commits/d4c12ec432512c9c1302c5fc2e8421ec2a6f8562#diff-d21ac5941571e624a797890f07d97e64bec5852597750e3794ed3716b3c7d471R84